libs/server/Resp/Objects/HashCommands.cs

```diff
@@ -114,18 +123,17 @@ private unsafe bool HashSet<TGarnetApi>(int count, byte* ptr, HashOperation hop,
         /// <typeparam name="TGarnetApi"></typeparam>
         /// <param name="count"></param>
         /// <param name="ptr"></param>
-        /// <param name="op">HGET, HGETALL, HMGET, HRANDFIELD</param>
         /// <param name="storageApi"></param>
         /// <returns></returns>
-        private unsafe bool HashGet<TGarnetApi>(int count, byte* ptr, HashOperation op, ref TGarnetApi storageApi)
+        private unsafe bool HashGet<TGarnetApi>(RespCommand command, int count, byte* ptr, ref TGarnetApi storageApi)
             where TGarnetApi : IGarnetApi
         {
-            if ((op == HashOperation.HGETALL && count != 1) ||
-                (op == HashOperation.HRANDFIELD && count < 1) ||
-                (op == HashOperation.HGET && count != 2) ||
-                (op == HashOperation.HMGET && count < 2))
+            if ((command == RespCommand.HGETALL && count != 1) ||
```

Comment:
HashOperation type should be passed as parameter and any ACL checks or mapping should happen at an earlier point.

Reply:
I want to push back gently on this (ow that ACL checks are all DRY'd up).  The command name ends up serialized in the failure cases, so we have a number of options here:

 1. Don't include the command name in error message, just pass `HashOperation`.
     * This is sort a reduction in quality but it is generic errors are used in some places already.
 2. Pass just the `RespCommand` and lookup the `HashOperation` in the method.
     * This is what I'm currently doing.
 3. Pass just the `HashOperation` and lookup the `RespCommand` on failure.
     * This feels backwards, but is about equivalent to 2.
 4. Pass just the `HashOperation` and assume its name matches `RespCommand`.
     * This is what the old code did, and feels wrong.  There's an implicit contract here that's pretty easy to accidentally break.
 5. Pass both the `HashOperation` and the `RespCommand`.
     * This feels redundant, and adds the chance of calling with mismatched parameters.  We could add a check to avoid that, but then we're doing the lookup anyway... so might as well do 2 or 3.

Of these I clearly then number 2 is most appropriate.

I don't feel that strongly, but I would like to hear why a different option is preferrable.

Reply:
> I was leaning towards 4 because write now we are doing one extra conversion from RespCommand to HashOperation.
I am not sure how things could break unless you are talking about adding more commands in this method which is not what I would recommend going forward
It is unfortunate that this method includes all HGET variants, it should be one NetworkOp method for each command.
This way we won't need have this conversion issue.
I believe with some refactoring work that we are doing this will eventually change in the future.
Overall, I don't feel too strongly against leaving it as is as long as we restructure it in the future.
Maybe others can chime in about it.

<sub><i>Moving above to this thread to keep context</i></sub>

The breaks I imagined were either adding new methods (which is admittedly rare, but Redis has like... 300 commands?  it's not impossible), or refactors that change the names of `HashOperation`s.

I do agree that changing these to one method per operation is the cleanest option.